tests/integration_test.go

```diff
@@ -758,7 +731,6 @@ func (s *IntegrationTestSuite) testErrorShouldOnlyTakeOfferOnce(asset types.EthA
 
 			if status != types.CompletedSuccess {
 				errCh <- fmt.Errorf("0th swap did not exit successfully: got %s", status)
-				cancel()
```

Comment:
I think this `cancel()` is the cause of the race condition we have in `TestError_ShouldOnlyTakeOfferOnce`. We were only calling `cancel()` here on Bob's swapd. The identical code for Charlie isn't calling it. The switch statement starting on line 798 below is handling `errCh` before `ctx.Done()` so I'm surprised that this was an issue. I would have expected the `errCh` case to get handled and the `ctx.Done()` case to be ignored.